lib/ex_doc/language/elixir.ex

```diff
@@ -0,0 +1,240 @@
+defmodule ExDoc.Language.Elixir do
+  @moduledoc false
+
+  @behaviour ExDoc.Language
+
+  alias ExDoc.Autolink
+
+  @impl true
+  def module_data(module) do
+    {type, skip} = module_type_and_skip(module)
+
+    %{
+      id: inspect(module),
+      title: module_title(module, type),
+      type: type,
+      skip: skip,
+      extra_callback_types: [:macrocallback]
+    }
+  end
+
+  @impl true
+  def function_data(entry, module_data) do
+    {{kind, name, arity}, _anno, _signature, _doc_content, metadata} = entry
+
+    extra_annotations =
+      case {kind, name, arity} do
+        {:macro, _, _} -> ["macro"]
+        {_, :__struct__, 0} -> ["struct"]
+        _ -> []
+      end
+
+    actual_def = actual_def(name, arity, kind)
+
+    %{
+      doc_fallback: fn ->
+        impl = Map.fetch(module_data.impls, actual_def)
+
+        callback_doc_ast(name, arity, impl) ||
+          delegate_doc_ast(metadata[:delegate_to])
+      end,
+      extra_annotations: extra_annotations,
+      line_override: find_function_line(module_data, actual_def),
```

Comment:
Do we need to call it override? Instead, should we simply say `line` is optional? And if one is not set, we default to one? Or rather, maybe we should be explicit and always ask for a `line`?

The other thing is that we have two entries: doc_line and source_line. I assume this one is source_line?

Reply:
For now I'll just rename this to `line`. About `find_module_line`, good point too, you're right that most likely on Erlang & Elixir will rely on it, so perhaps we should move that share functionality between each other, I'll add a TODO.